src/codegen/encoding/mod.rs

```diff
@@ -10,27 +10,526 @@ use crate::codegen::encoding::scale_encoding::ScaleEncoding;
 use crate::codegen::expression::load_storage;
 use crate::codegen::vartable::Vartable;
 use crate::codegen::{Builtin, Expression};
-use crate::sema::ast::{ArrayLength, Namespace, RetrieveType, StructType, Type};
+use crate::sema::ast::{ArrayLength, Namespace, RetrieveType, StructType, Type, U32};
 use crate::Target;
 use num_bigint::BigInt;
 use num_integer::Integer;
 use num_traits::{One, Zero};
 use solang_parser::pt::Loc;
 use std::ops::{AddAssign, MulAssign, Sub};
 
-/// This trait should be implemented by all encoding methods (ethabi, Scale and Borsh), so that
+pub(super) const LOC: Loc = Loc::Codegen;
```

Comment:
`LOC_CODEGEN` is barely shorter but even more awkward to read (to me at least) than `Loc::Codegen`.  The only point of this change is to make everything a bit less verbose.

Reply:
Ah, when you are writing `LOC`, it is not clear what variant of `Loc` you are referring to.

Reply:
> Here is an idea.
> 
> ```rust
> use solang_parser::pt::Loc::Codegen;
> ```
> 
> Now you can use `Codegen` rather than `Loc::Codegen`, and you're depending on the rust type system for the shorter name.

Oh, very nice, thanks!